src/Neo/SmartContract/Native/TransactionState.cs

```diff
@@ -70,6 +80,7 @@ void IInteroperable.FromStackItem(StackItem stackItem)
 
         StackItem IInteroperable.ToStackItem(ReferenceCounter referenceCounter)
         {
+            if (Transaction is null) return new Struct(referenceCounter) { new VM.Types.Array(referenceCounter, ConflictingSigners.Select(u => new ByteString(u.ToArray())).ToArray()) };
```

Comment:
Does we need to expose it? maybe it can be read by the smart contract and is not deterministic because it comes from the mempool not from the consensus

Reply:
> is not deterministic because it comes from the mempool not from the consensus

`ConflictingSigners` field of `TransactionState` actually *is* deterministic and has the same value within all blockchain  nodes, that's why we include it into the native Ledger contract storage and into the transaction state. `ConflictingSigners` field comes not from mempool, it comes from the persisted (in-block) transaction attributes.

> Does we need to expose it?
Well, the only user of `ConflictingSigners` field is the native Ledger contract, and it should be able to retrieve it. I'm not sure we can make it private.